### PR TITLE
Feature/player

### PR DESCRIPTION
**Pull request by racheljadel**
- Implements pause functionality for different classes (eg. UpgradeTree, ShipInteractionPopup etc.)
- Added/modified unit tests for several classes
- Added weapons to SaveableComponent
- Created audio for weapons
- Fixed code smells

### CODE REVIEW

File: saves/savefile.json
Comment (zombiechook) [inline saves/savefile.json:1]: There's merge conflicts here that need fixing

File: source/core/assets/levels/blazes_refuge/main_area/main.json
Comment (zombiechook) [inline source/core/assets/levels/blazes_refuge/main_area/main.json:1]: Shouldn't the map name on this be Blazes Refuge?

**Review by zombiechook — CHANGES_REQUESTED** [review]
savefile.json has a lot of merge conflicts that need fixing

File: source/core/assets/levels/blazes_refuge/main_area/main.json
Comment (racheljadel) [reply source/core/assets/levels/blazes_refuge/main_area/main.json:1]: That would make sense... for some reason in main it's still set at "Hell" for this file? I'll change it though, I don't believe there are any references to it as "Hell" so it shouldn't impact it.

File: saves/savefile.json
Comment (racheljadel) [reply saves/savefile.json:1]: Completely forgot about saves, thank you !

**Review by zombiechook — APPROVED** [review]
All good now

**Pull request merged**